core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -1608,18 +1639,26 @@ oppia.factory('explorationGadgetsService', [
 ]);
 
 // A service that returns the frontend representation of a newly-added state.
-oppia.factory('newStateTemplateService', [function() {
-  return {
-    // Returns a template for the new state with the given state name, changing
-    // the default rule destination to the new state name in the process.
-    // NB: clients should ensure that the desired state name is valid.
-    getNewStateTemplate: function(newStateName) {
-      var newStateTemplate = angular.copy(GLOBALS.NEW_STATE_TEMPLATE);
-      newStateTemplate.interaction.default_outcome.dest = newStateName;
-      return newStateTemplate;
-    }
-  };
-}]);
+oppia.factory('newStateTemplateService',
```

Comment:
You know, thinking about it, this should really be part of StateObjectFactory -- i.e. StateObjectFactory.createNewState(...). But we can leave this for a separate refactor, if you like. Maybe add a TODO though?